wire-ios-sync-engine/Source/Use cases/E2EIdentityCertificateUpdateStatusUseCase.swift

```diff
@@ -0,0 +1,119 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public enum E2EIdentityCertificateUpdateStatus {
+    case noAction, reminder, block
+}
+
+// sourcery: AutoMockable
+public protocol E2EIdentityCertificateUpdateStatusProtocol {
+    func invoke() async throws -> E2EIdentityCertificateUpdateStatus
+}
+public extension Notification.Name {
+    static let checkForE2EICertificateStatus = NSNotification.Name("CheckForE2EICertificateStatus")
+}
+
+final public class E2EIdentityCertificateUpdateStatusUseCase: E2EIdentityCertificateUpdateStatusProtocol {
+
+    private let e2eCertificateForCurrentClient: GetE2eIdentityCertificatesUseCaseProtocol
+    private let gracePeriod: TimeInterval
+    private let comparedDate: Date
+    private let lastAlertDate: Date?
+    private let mlsGroupID: MLSGroupID
+    private let mlsClientID: MLSClientID
+
+    public init(
+        e2eCertificateForCurrentClient: GetE2eIdentityCertificatesUseCaseProtocol,
+        gracePeriod: TimeInterval,
+        mlsGroupID: MLSGroupID,
+        mlsClientID: MLSClientID,
+        lastAlertDate: Date?,
+        comparedDate: Date = Date.now
+    ) {
+        self.e2eCertificateForCurrentClient = e2eCertificateForCurrentClient
+        self.gracePeriod = gracePeriod
+        self.lastAlertDate = lastAlertDate
+        self.mlsGroupID = mlsGroupID
+        self.mlsClientID = mlsClientID
+        self.comparedDate = comparedDate
+    }
+
+    @MainActor
```

Comment:
There is no point in putting a MainActor attribute on a use case method. Please remove!